pilot/util/harvester.py

```diff
@@ -244,19 +256,34 @@ def publish_work_report(work_report=None, worker_attributes_file="worker_attribu
 
     :param work_report: work report dictionary.
     :param worker_attributes_file:
-    :return:
+    :raises FileHandlingFailure: in case of IOError.
+    :return: True or False
     """
 
     if work_report:
-        work_report['timestamp'] = time_stamp()
-        if "outputfiles" in work_report:
-            del(work_report["outputfiles"])
-        if "inputfiles" in work_report:
-            del (work_report["inputfiles"])
-        if "xml" in work_report:
-            del (work_report["xml"])
-        if write_json(worker_attributes_file, work_report):
-            logger.info("work report published: {0}".format(work_report))
+        try:
+            work_report['timestamp'] = time_stamp()
+            if "outputfiles" in work_report:
+                del(work_report["outputfiles"])
+            if "inputfiles" in work_report:
+                del (work_report["inputfiles"])
+            if "xml" in work_report:
+                del (work_report["xml"])
+            if write_json(worker_attributes_file, work_report):
+                logger.info("work report published: {0}".format(work_report))
+                return True
+            else:
+                logger.info("work report publish failed: {0}".format(work_report))
+                return False
+        except IOError:
+            logger.error("job report copy failed")
+            return False
+        except:
```

Comment:
Blind exception, Flake8 complains - add change to 'except Exception as e:' and then logger.error('write json file failed: %s' % e)

Reply:
fixed

Reply:
fixed in now commit